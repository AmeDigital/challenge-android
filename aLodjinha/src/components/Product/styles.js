import { StyleSheet } from 'react-native';

import material from '../../utils/MaterialDesign';

export default StyleSheet.create({
	produto: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		paddingVertical: 15,
		paddingHorizontal: 5,
		borderBottomColor: material.colorGreyish,
		borderBottomWidth: 1
	},
	image: {
		width: 100,
		height: 100,
		marginRight: 5
	},
	containerDescription: {
		flex: 1,
		padding: 10
	},
	textName: {
		fontSize: 18,
		fontWeight: 'bold',
		color: material.colorDark,
		fontFamily: 'Roboto'
	},
	containerPrices: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		height: 40
	},
	priceOf: {
		color: material.colorGreyish,
		fontSize: 12,
		fontWeight: 'bold',
		textDecorationLine: 'line-through'
	},
	priceBy: {
		fontSize: 20,
		color: material.colorTomato,
		fontFamily: 'Roboto',
		fontWeight: 'bold'
	},
	containerImageRow: {
		width: 20,
		justifyContent: 'center',
		alignItems: 'center'
	},
	imageRow: {}
});
