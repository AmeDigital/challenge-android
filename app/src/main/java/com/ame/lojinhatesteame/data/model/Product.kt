package com.ame.lojinhatesteame.data.model

import android.text.Html
import com.ame.lojinhatesteame.util.formatToMoney
import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Product (
    @SerializedName("id") var id: Int = 0,
    @SerializedName("urlImagem") var urlImage: String = "",
    @SerializedName("nome") var name: String = "",
    @SerializedName("descricao") var description: String = "",
    @SerializedName("precoDe") var priceOf: Double = 0.0,
    @SerializedName("precoPor") var priceFor: Double = 0.0,
    @SerializedName("categoria") var category:Category? = null) : Serializable {

    val priceForFormated: String get() = "de ${priceFor.formatToMoney()}"
    val priceOfFormated: String get() = "por ${priceOf.formatToMoney()}"
    val descriptionHtmlFormat: String get() = Html.fromHtml(description).toString()
}