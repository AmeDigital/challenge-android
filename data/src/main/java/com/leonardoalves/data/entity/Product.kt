package com.leonardoalves.data.entity

import com.google.gson.annotations.SerializedName

data class Product(
	@SerializedName("id") val id: Int = 0,
	@SerializedName("nome") val nome: String? = null,
	@SerializedName("precoDe") val precoDe: Float? = null,
	@SerializedName("precoPor") val precoPor: Float? = null,
	@SerializedName("urlImagem") val urlImagem: String? = null,
	@SerializedName("descricao") val descricao: String? = null,
	@SerializedName("categoria") val categoria: Category? = null
)