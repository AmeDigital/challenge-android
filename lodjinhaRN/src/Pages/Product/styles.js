import { StyleSheet } from 'react-native';

export const Styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FCFCFC'
  },
  imageContainer: {
    flex: 1,
  },
  image: {
    height: 200
  },
  titleContainer: {
    flex: 0,
    padding: 16,
    borderBottomWidth: 1,
    borderBottomColor: '#d5d5d5'
  },
  title: {
    fontSize: 18,
    fontFamily: 'Roboto',
    fontWeight: 'bold'
  },
  description: {
    width: 100,
    height: 100,
    fontFamily: 'Roboto',
    fontSize: 40
  },
  priceContainer: {
    flex: 0,
    padding: 16,
    borderBottomWidth: 1,
    borderBottomColor: '#d5d5d5',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-end'
  },
  descriptionContainer: {
    flex: 1,
    padding: 16,
    marginBottom: 72
  },
  oldPrice: {
    fontFamily: 'Roboto',
    color: '#aaaaaa',
    textDecorationLine: 'line-through'
  },
  currentPrice: {
    fontFamily: 'Roboto',
    fontSize: 20,
    fontWeight: 'bold',
    color: '#f15025'
  },
})