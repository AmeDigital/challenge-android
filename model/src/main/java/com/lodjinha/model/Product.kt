package com.lodjinha.model

import com.google.gson.annotations.SerializedName

data class Product(
    val id: Int,
    @SerializedName("nome") val name: String,
    @SerializedName("descricao") val description: String,
    @SerializedName("urlImagem") val urlImage: String,
    @SerializedName("precoDe") val originalPrice: Double,
    @SerializedName("precoPor") val currentPrice: Double,
    @SerializedName("categoria") val category: Category
)
