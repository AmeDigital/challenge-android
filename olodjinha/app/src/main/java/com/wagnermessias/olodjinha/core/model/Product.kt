package com.wagnermessias.olodjinha.core.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Product(
    @SerializedName("id")
    val id: Int,

    @SerializedName("nome")
    val name: String,

    @SerializedName("urlImagem")
    val urlImage: String,

    @SerializedName("descricao")
    val description: String,

    @SerializedName("precoDe")
    val priceFrom: Float,

    @SerializedName("precoPor")
    val pricePer: Float,

    @SerializedName("categoria")
    val category: Category
): Serializable