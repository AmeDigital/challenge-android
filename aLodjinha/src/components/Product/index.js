import React from 'react';
import { View, Image, Text, TouchableOpacity } from 'react-native';

import styles from './styles';
import { replacePonitForComma } from '../../utils/Utils';

export default function Product({ item, funcPage }) {
	return (
		<View style={styles.produto}>
			<Image
				style={styles.image}
				source={{
					uri: item.urlImagem
				}}
			/>
			<View style={styles.containerDescription}>
				<Text style={styles.textDescription} onPress={() => funcPage(item.id)} numberOfLines={2}>
					{item.descricao}
				</Text>
				<View style={styles.containerPrices}>
					<Text style={styles.priceOf}>{`De: ${replacePonitForComma(item.precoDe)}`}</Text>
					<Text style={styles.priceBy}>{`Por ${replacePonitForComma(item.precoPor)}`}</Text>
				</View>
			</View>
			<View style={styles.containerImageRow}>
				<TouchableOpacity onPress={() => funcPage(item.id)}>
					<Image style={styles.imageRow} source={require('../../images/disclosure_indicator.png')} />
				</TouchableOpacity>
			</View>
		</View>
	);
}
