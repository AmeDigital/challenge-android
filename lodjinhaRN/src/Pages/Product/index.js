import React, { Component } from 'react';
import { View, ScrollView, Image, Text, SafeAreaView, Alert } from 'react-native';
import HTML from 'react-native-render-html';
import ActionButton from 'react-native-action-button';

import { formatNumber } from '../../helper';
import { Styles } from './styles';

export default class Product extends Component {
  static navigationOptions = {
    headerStyle: {
      elevation: 0,
      shadowOpacity: 0,
      borderBottomWidth: 0,
      backgroundColor: '#FCFCFC'
    }
  }

  handleClick = async (product) => {
    const response = await fetch(`https://alodjinha.herokuapp.com/produto/${product.id}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      }
    });
    const body = await response.json();
    
    if(body && body.result === 'success') {
      Alert.alert('', 'Produto reservado com sucesso!', [
        {text: 'ok' , onPress: () => this.props.navigation.goBack() }
      ])
    }
  }
    

  render() {
    const { product } = this.props.navigation.state.params;
    return (
      <SafeAreaView style={Styles.container}>
        <ScrollView style={Styles.container}>
          <View style={Styles.imageContainer}>
            <Image style={Styles.image} resizeMode="contain" source={{ uri: product.urlImagem }} />
          </View>
          <View style={Styles.titleContainer}>
            <Text style={Styles.title}>{product.nome}</Text>
          </View>
          <View style={Styles.priceContainer}>
            <Text style={Styles.oldPrice}>De: {formatNumber(product.precoDe)}</Text>
            <Text style={Styles.currentPrice}>Por {formatNumber(product.precoPor)}</Text>
          </View>
          <View style={Styles.descriptionContainer}>
            <HTML html={product.descricao} />
          </View>
        </ScrollView>
        <ActionButton 
        buttonColor="#5e2a84" 
        renderIcon={() => <Image style={{ width: 30, height: 20}} source={require('../../assets/images/check_button.png')} />} 
        onPress={async () => await this.handleClick(product)}></ActionButton>
      </SafeAreaView>
    );
  }
}
