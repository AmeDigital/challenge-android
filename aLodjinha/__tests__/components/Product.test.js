import React from 'react';

import renderer from 'react-test-renderer';

import Product from '../../src/components/Product';

describe('Component Product', () => {
	test('should render without erros', () => {
		const tree = renderer.create(<Product item={{ urlImagem: '' }} funcPage={function() {}} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
