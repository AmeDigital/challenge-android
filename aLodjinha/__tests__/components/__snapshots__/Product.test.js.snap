// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Product should render without erros 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#aaaaaa",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 15,
    }
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "height": 100,
        "marginRight": 5,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "padding": 10,
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        Object {
          "color": "#2d3142",
          "fontFamily": "Roboto",
          "fontSize": 35,
          "fontWeight": "bold",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#aaaaaa",
            "fontSize": 24,
            "fontWeight": "bold",
            "textDecorationLine": "line-through",
          }
        }
      >
        De: undefined
      </Text>
      <Text
        style={
          Object {
            "color": "#f15025",
            "fontFamily": "Roboto",
            "fontSize": 39,
            "fontWeight": "bold",
          }
        }
      >
        Por undefined
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 20,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/disclosure_indicator.png",
        }
      }
      style={Object {}}
    />
  </View>
</View>
`;
